internal/static/integrations/ssh_exporter/ssh_collector.go

```diff
@@ -0,0 +1,119 @@
+package ssh_exporter
+
+import (
+    "fmt"
+    "regexp"
+    "strconv"
+    "strings"
+
+    "github.com/go-kit/log"
+    "github.com/go-kit/log/level"
+    "github.com/prometheus/client_golang/prometheus"
+)
+
+type SSHClientInterface interface {
+    RunCommand(command string) (string, error)
+}
+
+type SSHCollector struct {
+    logger  log.Logger
+    target  Target
+    client  SSHClientInterface
+    metrics map[string]*prometheus.Desc
+}
+
+func NewSSHCollector(logger log.Logger, target Target) (*SSHCollector, error) {
+    client, err := NewSSHClient(target)
+    if err != nil {
+        return nil, err
+    }
+    client.logger = logger
+
+    collector := &SSHCollector{
+        logger:  logger,
+        target:  target,
+        client:  client,
+        metrics: make(map[string]*prometheus.Desc),
+    }
+
+    // Initialize metric descriptors for custom metrics
+    for _, cm := range target.CustomMetrics {
+        var labels []string
+        for label := range cm.Labels {
+            labels = append(labels, label)
+        }
+        desc := prometheus.NewDesc(cm.Name, cm.Help, labels, nil)
+        collector.metrics[cm.Name] = desc
+    }
+
+    return collector, nil
+}
+
+func (c *SSHCollector) Describe(ch chan<- *prometheus.Desc) {
+    for _, desc := range c.metrics {
+        ch <- desc
+    }
+}
+
+func (c *SSHCollector) Collect(ch chan<- prometheus.Metric) {
+    for _, cm := range c.target.CustomMetrics {
+        value, err := c.executeCustomCommand(cm)
+        if err != nil {
+            level.Error(c.logger).Log("msg", "failed to execute custom command", "command", cm.Command, "err", err)
+            continue
+        }
+
+        level.Debug(c.logger).Log("msg", "executed custom command", "command", cm.Command, "value", value)
```

Comment:
We could add debug statements like this to [Live Debugging](https://grafana.com/docs/alloy/latest/troubleshoot/debug/#live-debugging-page). They can also include the output of the command.